File: tang-commons/src/main/java/com/tang/commons/enumeration/MessageType.java
package com.tang.commons.enumeration;

import java.util.Arrays;

import com.fasterxml.jackson.annotation.JsonValue;

/**
 * 消息类型枚举类
 *
 * @author Tang
 */
public enum MessageType {

    NOTICE("notice"),
    MESSAGE("message"),
    HEARTBEAT("heartbeat"),
    CHAT_MESSAGE("chat_message");

    private final String name;

    MessageType(String name) {
        this.name = name;
    }

    @JsonValue
    public String getName() {
        return name;
    }

    public static MessageType getMessageType(String name) {
        return Arrays.stream(MessageType.values())
            .filter(messageType -> messageType.getName().equals(name))
            .findFirst()
            .orElse(null);
    }

}

File: tang-commons/src/main/kotlin/com/tang/commons/enumeration/MessageType.kt
package com.tang.commons.enumeration

import com.fasterxml.jackson.annotation.JsonValue

/**
 * 消息类型枚举类
 *
 * @author Tang
 */
enum class MessageType(@JsonValue val value: String) {

    NOTICE("notice"),

    MESSAGE("message"),

    HEARTBEAT("heartbeat"),

    CHAT_MESSAGE("chat_message");

    companion object {

        /**
         * 根据 value 返回枚举类型
         *
         * @param value 消息类型
         * @return {@link MessageType} 消息类型名称
         */
        @JvmStatic
        fun getMessageType(value: String): MessageType? {
            return entries.find { it.value == value }
        }

    }

}
